src/checkLinuxWebAppDownDetector.ts

```diff
@@ -0,0 +1,215 @@
+        ext.outputChannel.appendLine(detectorOutput);
+
+        const detectorUri: string = `https://management.azure.com/subscriptions/${node.root.subscriptionId}/resourceGroups/${node.root.client.resourceGroup}/providers/Microsoft.Web/sites/${node.root.client.siteName}/detectors/${detectorId}`;
+        const requestOptions: WebResource & Partial<{ qs: queryString }> = new WebResource();
+
+        requestOptions.method = 'GET';
+        requestOptions.url = detectorUri;
+        requestOptions.qs = {
+            'api-version': "2015-08-01",
+            fId: "1",
+            btnId: "2",
+            inpId: "1",
+            val: "vscode",
+            startTime: deployResultTime.toISOString(),
+            endTime: new Date().toISOString()
+        };
+        await signRequest(requestOptions, node.root.credentials);
+
+        let detectorErrorMessage: string | undefined;
+        // wait 10 minutes for a response from the detector
+        const detectorTimeoutMs: number = Date.now() + 10 * 60 * 1000;
+        do {
+            if (Date.now() > detectorTimeoutMs) {
+                const noIssuesFound: string = `No critical issues found for web app "${node.root.client.siteName}".`;
+                ext.outputChannel.appendLine(noIssuesFound);
+                return undefined;
+            }
+            // update the new end time to be now
+            requestOptions.qs.endTime = new Date().toISOString();
+            detectorErrorMessage = await getLinuxDetectorError(requestOptions, deployResultTime, node.root.client.fullName);
+            // poll every minute
+            await delay(1000 * 60);
+        } while (!detectorErrorMessage);
+
+        await ext.ui.showWarningMessage(detectorErrorMessage, { title: 'Open in Portal' });
+        const portalDeeplink = `https://portal.azure.com/?websitesextension_ext=asd.featurePath%3Ddetectors%2F${detectorId}#@microsoft.onmicrosoft.com/resource/${node.root.client.id}/troubleshoot`;
+        await openUrl(portalDeeplink);
+        context.telemetry.properties.didClick = 'true';
+    });
+}
+
+async function signRequest(req: WebResource, cred: ServiceClientCredentials): Promise<void> {
+    await new Promise((resolve: () => void, reject: (err: Error) => void): void => {
+        cred.signRequest(req, (err: Error | undefined) => {
+            if (err) {
+                reject(err);
+            } else {
+                resolve();
+            }
+        });
+    });
+}
+
+async function getLinuxDetectorError(requestOptions: WebResource, deployResultTime: Date, siteName: string): Promise<string | undefined> {
+    const requestPromise = <(options: RequestOptions | string | URL) => Promise<IncomingMessage>><Function>requestP;
+    const detectorResponse: string = <string><unknown>(await requestPromise(requestOptions));
+    const responseJson: detectorResponseJSON = <detectorResponseJSON>JSON.parse(detectorResponse);
+
+    const datasets: detectorDataset[] = responseJson.properties.dataset;
```

Comment:
Please move all of this parsing logic to a separate file and add unit tests. It's the kind of low-level logic that's easy to mess up and hard to review